File: make_id_dict.py
#! python3
# Gets the card data from netrunnerdb and extracts a dictionary of card name
# to card code
import requests, json

def make_id_dict(url="https://netrunnerdb.com/api/2.0/public/cards"):

    card_id_dict = {}

    r = requests.get(url=url)
    cards = r.json()["data"]
    for card in cards:
        code = card["code"]
        # remove ' to align with clj implementation
        title = card["title"].replace("'","")
        card_id_dict[title] = code
    return card_id_dict

if __name__ == "__main__":
    card_id_dict = make_id_dict()
    with open("card_id_dict.json", 'w', encoding="utf-8") as f:
        f.write(json.dumps(card_id_dict))
    

File: make_id_dict.jl
#!/usr/bin/env julia
# Gets the card data from netrunnerdb and extracts a dictionary of card name
# to card code
import Requests, JSON

"Create a dictionary of card title to card id number by reading all cards on NRDB"
function make_id_dict(url="https://netrunnerdb.com/api/2.0/public/cards")
  card_id_dict = Dict()

  r = Requests.get(url)
  cards = Requests.json(r)["data"]
  for card in cards
    code = card["code"]
    # remove ' to align with clj implementation
    title = replace(card["title"], "'","")
    card_id_dict[title] = code
  end

  println("Created Dict(title => id) for $(length(cards)) cards")
  return card_id_dict
end

card_id_dict = make_id_dict()
open(f -> JSON.print(f, card_id_dict), "card_id_dict.json", "w")
